test/partition.test.js

```diff
@@ -0,0 +1,164 @@
+// Copyright IBM Corp. 2016,2019. All Rights Reserved.
+// Node module: loopback-connector-cloudant
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+require('./init.js');
+const _ = require('lodash');
+const should = require('should');
+const DEFAULT_MODEL_VIEW = 'loopback__model__name';
+let Product, db, connector;
+
+const config = {
+  url: process.env.CLOUDANT_URL,
+  username: process.env.CLOUDANT_USERNAME,
+  password: process.env.CLOUDANT_PASSWORD,
+  database: process.env.CLOUDANT_PARTITIONED_DATABASE,
+  plugin: 'retry',
+  retryAttempts: 10,
+  retryTimeout: 50,
+};
+
+describe('cloudant - partitioned db', () => {
+  before(function(done) {
+    db = global.getDataSource(config);
+    connector = db.connector;
+    db.automigrate(done);
+  });
+
+  it('default - create global index for property', (done) => {
+    Product = db.define('Product', {
+      prodName: {type: String, index: true},
+      desc: {type: String},
+    });
+    db.autoupdate('Product', (err) => {
+      if (err) return done(err);
+      connector.getIndexes(connector.getDbName(connector), (e, results) => {
+        if (e) return done(e);
+        const indexes = results.indexes;
+        const indexName = 'prodName_index';
+        should.exist(indexes);
+
+        const index = _.find(indexes, function(index) {
+          return index.name === indexName;
+        });
+        should.exist(index);
+        index.name.should.equal(indexName);
+        index.def.fields[0]['prodName'].should.equal('asc');
+        index.def.fields[1][DEFAULT_MODEL_VIEW].should.equal('asc');
+        // should be a global index
+        index.partitioned.should.equal(false);
+        done();
+      });
+    });
+  });
+
+  it('default - create global index for index entry', (done) => {
+    Product = db.define('Product', {
+      prodName: {type: String},
+      desc: {type: String},
+    }, {
+      indexes: {
+        'prodName1_index': {
+          keys: {
+            prodName: -1,
+          },
+        },
+      },
+    });
+    db.autoupdate('Product', (err) => {
+      if (err) return done(err);
+      connector.getIndexes(connector.getDbName(connector), (e, results) => {
+        if (e) return done(e);
+        const indexes = results.indexes;
+        const indexName = 'prodName1_index';
+        should.exist(indexes);
+
+        const index = _.find(indexes, function(index) {
+          return index.name === indexName;
+        });
+        should.exist(index);
+        index.name.should.equal(indexName);
+        index.def.fields[0]['prodName'].should.equal('desc');
+        index.def.fields[1][DEFAULT_MODEL_VIEW].should.equal('desc');
```

Comment:
I am not familiar with Cloudant index field. Why is this `desc` and others are `asc`?

Reply:
@agnes512 Good question, see https://github.com/strongloop/loopback-connector-cloudant/pull/225/files#diff-121c1d00e6380aa6109385e1df1ae803R66

The order code specified for `prodName` is -1, which stands for `desc`

Reply:
looks like they need a better name :laughing: